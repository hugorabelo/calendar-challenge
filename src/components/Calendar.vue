<template>
  <div class="container mx-auto">
    <calendar-header @today="today()" @previousMonth="previousMonth()" @nextMonth="nextMonth()"></calendar-header>
    <div class="grid grid-cols-7 border border-gray-300 h-screen">
      <day v-for="(day, index) in days" :fullDate="day" :key="index"></day>
    </div>
  </div>
</template>

<script>
import { Interval } from 'luxon'
import CalendarHeader from './CalendarHeader'
import Day from './Day'
import { mapActions, mapGetters } from 'vuex'

export default {
  name: 'Calendar',
  components: {
    CalendarHeader,
    Day,
  },
  data() {
    return {
      days: [],
    }
  },
  computed: {
    ...mapGetters(['selectedMonth']),
  },
  mounted() {
    this.fetchDaysOfSelectedMonth()
  },
  methods: {
    ...mapActions(['initSelectedMonth', 'goToNextMonth', 'goToPreviousMonth']),
    fetchDaysOfSelectedMonth() {
      this.days = []
      let firstDayOfMonth = this.selectedMonth.startOf('month')
      let lastDayOfMonth = this.selectedMonth.endOf('month')
      let dayOfWeek = firstDayOfMonth.toFormat('c')
      var currentDay

      if (dayOfWeek != 7) {
        for (var i = dayOfWeek; i > 0; i--) {
          currentDay = firstDayOfMonth.minus({ days: i })
          this.days.push(currentDay)
        }
      }

      currentDay = firstDayOfMonth
      while (Interval.fromDateTimes(currentDay, lastDayOfMonth).length('days') > 0) {
        this.days.push(currentDay)
        currentDay = currentDay.plus({ days: 1 })
      }
      dayOfWeek = currentDay.toFormat('c')
      if (dayOfWeek != 7) {
        for (var j = 0; j < 7 - dayOfWeek; j++) {
          this.days.push(currentDay)
          currentDay = currentDay.plus({ days: 1 })
        }
      }
    },
    today() {
      this.initSelectedMonth()
      this.fetchDaysOfSelectedMonth()
    },
    previousMonth() {
      this.goToPreviousMonth()
      this.fetchDaysOfSelectedMonth()
    },
    nextMonth() {
      this.goToNextMonth()
      this.fetchDaysOfSelectedMonth()
    },
  },
}
</script>
