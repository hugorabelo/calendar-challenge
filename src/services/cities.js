import axios from 'axios'
import httpAdapter from 'axios/lib/adapters/http'

const API_CITY_KEY = process.env.VUE_APP_API_CITY_KEY

const http = axios.create({
  baseURL: 'https://andruxnet-world-cities-v1.p.rapidapi.com/',
  adapter: httpAdapter,
  headers: {
    'X-RapidAPI-Host': 'andruxnet-world-cities-v1.p.rapidapi.com',
    'X-RapidAPI-Key': API_CITY_KEY,
  },
})

export function getCities(partial) {
  return http.get(`?query=${partial}&searchby=city`)
}
