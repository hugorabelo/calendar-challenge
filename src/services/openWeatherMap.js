import axios from 'axios'
import httpAdapter from 'axios/lib/adapters/http'

const API_KEY = '0d8aa4b5581a5331b53fe1c53c106c4f'

const http = axios.create({
  baseURL: 'https://api.openweathermap.org/data/2.5/',
  adapter: httpAdapter,
})

export async function getWeatherForecast(city) {
  let excluded = 'current,alerts,minutely,hourly'
  return await http.get(`onecall?lat=${city.coord.lat}&lon=${city.coord.lon}&exclude=${excluded}&appid=${API_KEY}`)
}

export function getWeatherForecastByGeoLocation(geoLocation) {
  let excluded = 'current,alerts,minutely,hourly'
  return http.get(`onecall?lat=${geoLocation.lat}&lon=${geoLocation.lon}&exclude=${excluded}&appid=${API_KEY}`)
}

export function getCityGeoLocation(city) {
  return http.get(`weather?q=${city}&APPID=${API_KEY}`)
}
