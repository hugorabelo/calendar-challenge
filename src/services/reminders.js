import store from '@/store'

export function getReminders() {
  return store.getters.reminders
}

export function getReminderById(id) {
  let selectedReminder = store.getters.reminders.filter((rem) => rem.id == id)
  return selectedReminder.length > 0 ? selectedReminder[0] : null
}

export function addReminder(data) {
  store.dispatch('addReminder', data)
}

export function updateReminder(data) {
  store.dispatch('updateReminder', data)
}

export function deleteReminder(data) {
  store.dispatch('deleteReminder', data)
}

export function deleteRemindersGroup(data) {
  store.dispatch('deleteRemindersGroup', data)
}
