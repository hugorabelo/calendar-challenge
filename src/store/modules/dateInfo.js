import { DateTime } from 'luxon'

const state = {
  selectedMonth: DateTime.now().startOf('month'),
}
const getters = {
  selectedMonth: (state) => state.selectedMonth,
}
const mutations = {
  setSelectedMonth(state, data) {
    state.selectedMonth = data
  },
}
const actions = {
  initSelectedMonth({ commit }) {
    commit('setSelectedMonth', DateTime.now().startOf('month'))
  },
  goToNextMonth({ commit, state, dispatch }) {
    if (!state.selectedMonth) {
      dispatch('initSelectedMonth')
    }
    let nextMonth = state.selectedMonth.plus({ months: 1 })
    commit('setSelectedMonth', nextMonth)
  },
  goToPreviousMonth({ commit, state, dispatch }) {
    if (!state.selectedMonth) {
      dispatch('initSelectedMonth')
    }
    let previousMonth = state.selectedMonth.minus({ months: 1 })
    commit('setSelectedMonth', previousMonth)
  },
}

export default {
  state,
  getters,
  actions,
  mutations,
}
