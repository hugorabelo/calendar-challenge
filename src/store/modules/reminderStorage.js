const state = {
  reminders: localStorage.getItem('reminders') != null ? JSON.parse(localStorage.getItem('reminders')) : [],
}
const getters = {
  reminders: (state) => state.reminders,
}
const mutations = {
  setReminders(state, data) {
    state.reminders = data
    localStorage.setItem('reminders', JSON.stringify(state.reminders))
  },
}
const actions = {
  addReminder({ commit, state }, data) {
    if (state.reminders.filter((rem) => rem.id == data.id).length == 0) {
      state.reminders.push(data)
    }
    commit('setReminders', state.reminders)
  },
  updateReminder({ commit, state }, data) {
    for (var i = 0; i < state.reminders.length; i++) {
      if (state.reminders[i].id == data.id) {
        state.reminders[i] = data
      }
    }
    commit('setReminders', state.reminders)
  },
  deleteReminder({ commit, state }, data) {
    for (var i = 0; i < state.reminders.length; i++) {
      if (state.reminders[i].id == data.id) {
        state.reminders.splice(i, 1)
      }
    }
    commit('setReminders', state.reminders)
  },
  deleteRemindersGroup({ dispatch }, data) {
    for (var i = 0; i < data.length; i++) {
      dispatch('deleteReminder', { id: data[i].id })
    }
  },
}

export default {
  state,
  getters,
  actions,
  mutations,
}
