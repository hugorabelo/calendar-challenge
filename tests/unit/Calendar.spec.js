import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Calendar from '@/components/Calendar'
import { DateTime } from 'luxon'
import dateInfo from '@/store/modules/dateInfo'
import reminderStorage from '@/store/modules/reminderStorage'
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
import { library } from '@fortawesome/fontawesome-svg-core'
import {
  faBan,
  faCalendar,
  faChevronLeft,
  faChevronRight,
  faEdit,
  faLocationDot,
  faNoteSticky,
  faPlus,
  faTrash,
  faXmark,
} from '@fortawesome/free-solid-svg-icons'
import flushPromisses from 'flush-promises'
import nock from 'nock'
import * as weatherApi from '@/services/openWeatherMap'
import weatherData from './fixtures/weatherData'
import cityData from './fixtures/cityData'

const localVue = createLocalVue()
localVue.use(Vuex)
library.add(
  faChevronLeft,
  faChevronRight,
  faPlus,
  faBan,
  faEdit,
  faTrash,
  faXmark,
  faNoteSticky,
  faCalendar,
  faLocationDot
)
localVue.component('font-awesome-icon', FontAwesomeIcon)

describe('Calendar', () => {
  const expectedDataCity = {
    q: 'London',
    APPID: '0d8aa4b5581a5331b53fe1c53c106c4f',
  }

  const expectedDataWeather = {
    lat: '51.5085',
    lon: '-0.1257',
    exclude: 'current,alerts,minutely,hourly',
    appid: '0d8aa4b5581a5331b53fe1c53c106c4f',
  }

  const reminder = {
    id: '123-456',
    description: 'Event 01',
    dateTime: DateTime.now().toISO(),
    city: 'London',
    color: 'bg-red-600',
    weatherForecast: 'light rain',
    weatherIcon: '10n',
  }

  const build = (options) => {
    return mount(Calendar, {
      localVue,
      store: new Vuex.Store({
        modules: {
          dateInfo: { ...dateInfo },
          reminderStorage: { ...reminderStorage },
        },
      }),
      ...options,
    })
  }

  test('Is a Vue instance and have days', () => {
    const { vm } = build()
    expect(vm).toBeTruthy()

    expect(vm.days.length).toBeGreaterThan(0)
  })

  test('Navigate by months', async () => {
    const wrapper = build()

    const store = new Vuex.Store({
      modules: {
        dateInfo: dateInfo,
      },
    })

    let currentMonth = DateTime.now().startOf('month')
    let nextMonth = currentMonth.plus({ months: 1 })
    let previousMonth = currentMonth.minus({ months: 1 })

    expect(wrapper.html()).toMatchSnapshot()

    let btnToday = wrapper.findComponent('#btn-today')
    let btnPrevious = wrapper.findComponent('#btn-previous')
    let btnNext = wrapper.findComponent('#btn-next')

    await btnToday.trigger('click')
    expect(currentMonth).toEqual(store.state.dateInfo.selectedMonth)

    await btnPrevious.trigger('click')
    expect(previousMonth).toEqual(store.state.dateInfo.selectedMonth)

    await btnNext.trigger('click')
    expect(currentMonth).toEqual(store.state.dateInfo.selectedMonth)

    await btnNext.trigger('click')
    expect(nextMonth).toEqual(store.state.dateInfo.selectedMonth)
  })

  test('First day of the month is Sunday', async () => {
    const store = new Vuex.Store({
      modules: {
        dateInfo: dateInfo,
      },
    })

    let currentMonth = DateTime.fromISO('2022-05-01')
    store.state.dateInfo.selectedMonth = currentMonth

    const { vm } = build()

    expect(vm.days.length).toBe(35)
  })

  test('Ability to add a new "reminder" for a user entered day and time. Also, include a city.', async () => {
    const wrapper = build()
    const requestCity = nock('https://api.openweathermap.org/data/2.5')
      .persist()
      .get('/weather')
      .query(expectedDataCity)
      .reply(200, cityData)

    const responseCity = await weatherApi.getCityGeoLocation(reminder.city)
    await flushPromisses()

    expect(responseCity.data).toEqual(cityData)
    expect(requestCity.isDone()).toBe(true)

    const requestWeather = nock('https://api.openweathermap.org/data/2.5')
      .persist()
      .get('/onecall')
      .query(expectedDataWeather)
      .reply(200, weatherData)

    const responseWeather = await weatherApi.getWeatherForecast(reminder.city)
    await flushPromisses()

    expect(responseWeather.data).toEqual(weatherData)
    expect(requestWeather.isDone()).toBe(true)

    expect(wrapper.html()).toMatchSnapshot()

    let btnAddReminder = wrapper.findComponent('#btn-add-reminder')
    await btnAddReminder.trigger('click')

    let btnCloseDialog = wrapper.findComponent('#btn-close-dialog')
    await btnCloseDialog.trigger('click')

    await btnAddReminder.trigger('click')

    let inputDescription = wrapper.findComponent('#input-description')
    let options = wrapper.findComponent('#input-city').findAll('option')
    // let reminderForm = wrapper.findComponent('reminder-form')
    let btnSave = wrapper.findComponent('#btn-save')

    await inputDescription.setValue(reminder.description)
    await options.at(-1).setSelected()

    await btnSave.trigger('click')
    expect(wrapper.emitted('save')).toBeTruthy()
  })
})
