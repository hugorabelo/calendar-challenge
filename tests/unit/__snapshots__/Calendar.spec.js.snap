// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Ability to add a new "reminder" for a user entered day and time. Also, include a city. 1`] = `
<div class="container mx-auto">
  <div class="header flex items-center p-4"><button id="btn-today" class="py-2.5 px-5 mr-2 text-sm font-medium text-gray-900 focus:outline-none bg-white rounded-lg border border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-700">
      Today
    </button> <button id="btn-previous" class="rounded-full h-10 w-10 hover:bg-gray-200 mr-1"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="chevron-left" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512" class="svg-inline--fa fa-chevron-left">
        <path fill="currentColor" d="M224 480c-8.188 0-16.38-3.125-22.62-9.375l-192-192c-12.5-12.5-12.5-32.75 0-45.25l192-192c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25L77.25 256l169.4 169.4c12.5 12.5 12.5 32.75 0 45.25C240.4 476.9 232.2 480 224 480z" class=""></path>
      </svg></button> <button id="btn-next" class="rounded-full h-10 w-10 hover:bg-gray-200 mr-2 md:mr-8"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="chevron-right" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512" class="svg-inline--fa fa-chevron-right">
        <path fill="currentColor" d="M96 480c-8.188 0-16.38-3.125-22.62-9.375c-12.5-12.5-12.5-32.75 0-45.25L242.8 256L73.38 86.63c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l192 192c12.5 12.5 12.5 32.75 0 45.25l-192 192C112.4 476.9 104.2 480 96 480z" class=""></path>
      </svg></button>
    <div class="text-3xl md:text-3xl hidden md:flex long-title">
      maio de 2022
    </div>
    <div class="md:text-3xl md:hidden short-title">
      mai. de 2022
    </div> <button id="btn-add-reminder" class="hidden md:flex mx-auto text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 mr-0 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800">
      Add New Reminder
    </button> <button type="button" class="mx-auto mr-0 md:hidden text-white border bg-blue-700 font-medium rounded-full text-sm p-2.5 text-center inline-flex items-center"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="plus" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="svg-inline--fa fa-plus">
        <path fill="currentColor" d="M432 256c0 17.69-14.33 32.01-32 32.01H256v144c0 17.69-14.33 31.99-32 31.99s-32-14.3-32-31.99v-144H48c-17.67 0-32-14.32-32-32.01s14.33-31.99 32-31.99H192v-144c0-17.69 14.33-32.01 32-32.01s32 14.32 32 32.01v144h144C417.7 224 432 238.3 432 256z" class=""></path>
      </svg></button>
  </div>
  <div class="grid grid-cols-7 border border-gray-300 h-screen-adjust">
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell bg-blue-50 hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          1
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          2
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          3
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          4
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          5
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          6
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell bg-blue-50 hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          7
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell bg-blue-50 hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          8
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          9
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          10
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          11
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          12
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          13
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell bg-blue-50 hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          14
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell bg-blue-50 hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          15
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          16
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          17
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          18
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          19
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          20
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell bg-blue-50 hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          21
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell bg-blue-50 hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          22
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          23
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          24
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          25
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          26
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          27
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell bg-blue-50 hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          28
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell bg-blue-50 hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          29
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          30
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell hover:border-blue-700">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day">
          31
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day text-gray-300">
          1
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day text-gray-300">
          2
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day text-gray-300">
          3
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="flex flex-col box-border border border-gray-300 md:p-4 p-1 day-cell bg-blue-50">
      <div class="relative w-full justify-items-center">
        <div class="relative h-8 text-lg text-black font-medium label-day text-gray-300">
          4
        </div>
        <div class="w-full flex flex-col">
          <div>
            <div class="w-full flex flex-col"></div>
            <!---->
          </div>
        </div>
      </div>
    </div>
  </div>
  <!---->
</div>
`;

exports[`Calendar Navigate by months 1`] = `
<div class="container mx-auto">
  <div class="header flex items-center p-4"><button id="btn-today" class="py-2.5 px-5 mr-2 text-sm font-medium text-gray-900 focus:outline-none bg-white rounded-lg border border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-700">
      Today
    </button> <button id="btn-previous" class="rounded-full h-10 w-10 hover:bg-gray-200 mr-1"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="chevron-left" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512" class="svg-inline--fa fa-chevron-left">
        <path fill="currentColor" d="M224 480c-8.188 0-16.38-3.125-22.62-9.375l-192-192c-12.5-12.5-12.5-32.75 0-45.25l192-192c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25L77.25 256l169.4 169.4c12.5 12.5 12.5 32.75 0 45.25C240.4 476.9 232.2 480 224 480z" class=""></path>
      </svg></button> <button id="btn-next" class="rounded-full h-10 w-10 hover:bg-gray-200 mr-2 md:mr-8"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="chevron-right" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512" class="svg-inline--fa fa-chevron-right">
        <path fill="currentColor" d="M96 480c-8.188 0-16.38-3.125-22.62-9.375c-12.5-12.5-12.5-32.75 0-45.25L242.8 256L73.38 86.63c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l192 192c12.5 12.5 12.5 32.75 0 45.25l-192 192C112.4 476.9 104.2 480 96 480z" class=""></path>
      </svg></button>
    <div class="text-3xl md:text-3xl hidden md:flex long-title">
      abril de 2022
    </div>
    <div class="md:text-3xl md:hidden short-title">
      abr. de 2022
    </div> <button id="btn-add-reminder" class="hidden md:flex mx-auto text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 mr-0 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800">
      Add New Reminder
    </button> <button type="button" class="mx-auto mr-0 md:hidden text-white border bg-blue-700 font-medium rounded-full text-sm p-2.5 text-center inline-flex items-center"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="plus" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="svg-inline--fa fa-plus">
        <path fill="currentColor" d="M432 256c0 17.69-14.33 32.01-32 32.01H256v144c0 17.69-14.33 31.99-32 31.99s-32-14.3-32-31.99v-144H48c-17.67 0-32-14.32-32-32.01s14.33-31.99 32-31.99H192v-144c0-17.69 14.33-32.01 32-32.01s32 14.32 32 32.01v144h144C417.7 224 432 238.3 432 256z" class=""></path>
      </svg></button>
  </div>
  <div class="grid grid-cols-7 border border-gray-300 h-screen-adjust"></div>
  <!---->
</div>
`;
